src/wallet/feebumper.cpp

```diff
@@ -70,11 +70,17 @@ static feebumper::Result PreconditionChecks(const CWallet* wallet, const CWallet
 
 namespace feebumper {
 
+bool TransactionCanBeReplaced(const CWalletTx& wtx)
+{
+    const int64_t replacement_timeout = gArgs.GetArg("-mempoolreplacementtimeout", DEFAULT_REPLACEMENT_TIMEOUT);
+    return ExpiredOptInRBFPolicy(GetTime(), wtx.nTimeReceived, replacement_timeout) || SignalsOptInRBF(*wtx.tx);
+}
+
 bool TransactionCanBeBumped(CWallet* wallet, const uint256& txid)
 {
     LOCK2(cs_main, wallet->cs_wallet);
     const CWalletTx* wtx = wallet->GetWalletTx(txid);
-    return wtx && SignalsOptInRBF(*wtx->tx) && !wtx->mapValue.count("replaced_by_txid");
+    return wtx && TransactionCanBeReplaced(*wtx) && !wtx->mapValue.count("replaced_by_txid");
```

Comment:
Dunno that we want to support replacing a transaction in wallet based on this. If nothing else it needs a buffer in excess of the replacement timeout to ensure propagation works well, and probably shouldn't let the user override the replacement timeout in wallet. May also be good to wait some time for network upgrade, ie amybe this should be a separate PR in a separate release.

Reply:
Fair enough, I'll move the wallet part to a new PR. What sort of buffer you had in mind?  a percentage of the replacement timeout, say 10%? or perhaps better to have fixed buffer of a few minutes?

Reply:
Just need something to capture the time for the original transaction to propagate. A few minutes is probably fine, but no reason not to be conservative...say, 20 minutes?

Reply:
Seems reasonable to me

Reply:
I think this is now all addressed